modules/file_writer_module/src/parquet_fb_impl.cpp

```diff
@@ -0,0 +1,368 @@
+#include <file_writer_module/builder_type_transit.h>
+
+#include <coreobjects/unit_factory.h>
+#include <coreobjects/eval_value_factory.h>
+
+#include <opendaq/input_port_factory.h>
+#include <opendaq/data_descriptor_ptr.h>
+#include <opendaq/event_packet_ptr.h>
+#include <opendaq/signal_factory.h>
+#include <opendaq/custom_log.h>
+#include <opendaq/event_packet_params.h>
+#include <opendaq/data_packet.h>
+#include <opendaq/data_packet_ptr.h>
+#include <opendaq/event_packet_ids.h>
+#include <opendaq/packet_factory.h>
+#include <opendaq/range_factory.h>
+#include <opendaq/sample_type_traits.h>
+#include <opendaq/reusable_data_packet_ptr.h>
+#include <opendaq/component_status_container_private_ptr.h>
+#include <coreobjects/validator_factory.h>
+
+
+BEGIN_NAMESPACE_FILE_WRITER_MODULE
+
+namespace FileWriter
+{
+
+ParquetFbImpl::ParquetFbImpl(const ContextPtr& ctx, const ComponentPtr& parent, const StringPtr& localId)
+    : FunctionBlock(CreateType(), ctx, parent, localId)
+    , inputPortCount(0)
+    , tableNumberCount(0)
+    , recordingActive(false)
+    , downScalingDevider(1)
+{
+    initProperties();
+    createAndAddInputPort(fmt::format("Input{}", inputPortCount++), PacketReadyNotification::SameThread);
+}
+
+void ParquetFbImpl::initProperties()
+{
+    path = "";
+    const auto pathProp = StringProperty("Path", path);
+    objPtr.addProperty(pathProp);
+    objPtr.getOnPropertyValueWrite("Path") +=
+        [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { propertyChanged(false); };
+
+    const auto fileNameProp = StringProperty("FileName", "ExampleFile");
+    objPtr.addProperty(fileNameProp);
+    objPtr.getOnPropertyValueWrite("FileName") +=
+        [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { propertyChanged(false); };
+
+    const auto recordingActiveProp = BoolProperty("RecordingActive", false);
+    objPtr.addProperty(recordingActiveProp);
+    objPtr.getOnPropertyValueWrite("RecordingActive") +=
+        [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { propertyActiveChanged(); };
+
+    const auto batchCylce = IntPropertyBuilder("BatchCylce", 20)
+                            .setUnit(Unit("s"))
+                            .setMinValue(1)
+                            .setMaxValue(3600)
+                            .setDescription("After how many seconds a batch is stored.")
+                            .build();
+    objPtr.addProperty(batchCylce);
+    objPtr.getOnPropertyValueWrite("BatchCylce") +=
+        [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { propertyChanged(true); };
+
+
+    const auto downScalingDevider = IntPropertyBuilder("DownScalingDevider", 1)
+                            .setMinValue(1)
+                            .setMaxValue(1000000)
+                            .setDescription("Only every nth value is stored to the file.")
+                            //.setValidator(Validator("$RecordingActive == False"))
```

Comment:
@JakaMohorkoDS : Why is the validator not working. I want to have the validation happen if the RecordingActive is True that this property cannot be manipulated.